server/src/Utils/Order.hs

```diff
@@ -0,0 +1,38 @@
+{-# LANGUAGE FlexibleContexts #-}
+{-# LANGUAGE NoImplicitPrelude #-}
+{-# LANGUAGE OverloadedStrings #-}
+{-# LANGUAGE MultiParamTypeClasses #-}
+{-# LANGUAGE TypeFamilies #-}
+
+module Utils.Order where
+
+import Import.NoFoundation
+import Model.Types
+
+-- {-| Determine if a given user is allowed to access the PDF file of an item.
+-- -}
+hasAccessToPdfFileDownload ::
+       ( BaseBackend (YesodPersistBackend site) ~ SqlBackend
+       , YesodPersist site
+       , PersistQueryRead (YesodPersistBackend site)
+       )
+    => UserId
+    -> Text
+    -> HandlerT site IO AuthResult
+hasAccessToPdfFileDownload userId filename = do
+    let unauthorized = Unauthorized "You didn't buy this item"
```

Comment:
Oh, those words! :) 

I actually haven't dealt with monad transformers yet (just know about them). Could you write a few lines of (non-compiling) code, just to give me a skeleton of what I'd be expected to do.